src/components/PasskeyPrf.jsx

```diff
@@ -0,0 +1,122 @@
+import { base64 } from "@scure/base";
+import { DID } from "dids";
+import { Ed25519Provider } from "key-did-provider-ed25519";
+import KeyResolver from "key-did-resolver";
+import { EncryptionScheme, atou, utoa } from "./EncryptionScheme.jsx";
+
+async function createCredential() {
+  const credential = await navigator.credentials.create({
+    publicKey: {
+      challenge: Uint8Array.from([0x5a, 0xfe]),
+      rp: {
+        name: "Web3 Encryption Experiments",
+        id: window.location.hostname,
+      },
+      user: {
+        id: Uint8Array.from([0x5a, 0xfe]),
+        name: "research@safe.dev",
+        displayName: "Researcher",
+      },
+      pubKeyCredParams: [
+        { alg: -7, type: "public-key" }, // ES256
+      ],
+      timeout: 60000,
+      attestation: "none",
+      extensions: {
+        prf: {},
+      },
+    },
+  });
+  if (!credential?.prf?.enabled) {
+    // On macOS Safari, the extension is not reported as enabled, even if it
+    // does reply with PRF entropy when attesting.
+    console.warn("PRF extension may not supported by the authenticator");
+  }
+}
+
+async function getCredential() {
+  const credential = await navigator.credentials.get({
+    publicKey: {
+      challenge: Uint8Array.from([0x5a, 0xfe]),
+      userVerification: "discouraged",
+      timeout: 60000,
+      extensions: {
+        prf: {
+          eval: {
+            first: Uint8Array.from([0x5a, 0xfe]),
+          },
+        },
+      },
+    },
+  });
+  return credential;
+}
+
+async function getPrf() {
+  let credential;
```

Comment:
```suggestion
async function getPrf() {
  // Check WebAuthn support first
  if (!window.PublicKeyCredential) {
    throw new Error("WebAuthn is not supported in this browser");
  }

  let credential;
```

It's mostly [compatible](https://developer.mozilla.org/en-US/docs/Web/API/PublicKeyCredential#browser_compatibility), so can ignore this suggestion.